rfcs/0000-c-unwind-and-project-group-charter.md

```diff
@@ -0,0 +1,496 @@
+Rust frames triggered unwinding mechanisms. Note that in general a
+longjmp over a Rust frame that contains destructors is [Undefined
+Behavior] -- however, we would like to support longjmp for Rust frames
+that do not contain destructors. At minimum, this group should take
+the behavior of longjmp into account; if it is easy to do, we may also
+introduce mechanisms or RFCs that help to specify the use of longjmp
+in particular cases.
+
+## Our proposal
+
+This RFC proposes a multi-prong plan to solve this problem:
+
+* First, to modify Rust functions with the "C" abi to abort on unwinding, as proposed
+  in [rust-lang/rust#52652].
+* Second, to introduce the "C unwind" ABI, with virtually all details
+  (but not quite all) left as "to be determined".
+    * The **intent** of this ABI, however, is to permit Rust panics to
+      be "translated" into a "native" unwinding mechanism, though
+      precisely which is dependent on the target platform or other
+      compiler options.
+        * (In practice, this translation is a no-op, since Rust panics
+          use the native mechanism, but this is explicitly not
+          required for the future.)
+    * Folks currently using the "C" ABI to do unwinding can migrate to this new
+      "C unwind" ABI, which captures their intentions. Their code is no more stable
+      than it was, but they are on the right path.
+* Finally, to create a "project group" with the purpose of designing subsequent
+  RFCs to flesh out the details of the "C unwind" ABI
+    * The "project group" term is newly introduced, but it corresponds
+      to a kind of working group -- one with the goal of fleshing out
+      a particular proposal or completing a project.
+    * We aim to specify how "C unwind" works on major platforms
+    * Goal is to enable Rust panics to propagate across native frames
+    * And perhaps to enable native exceptions to propagate across Rust frames
+    * But not to allow catching or throwing native exceptions from Rust code
+
+The first two goals work together. We simul
+The goal of introducing the "C unwind" ABI is simple: it allows
+existing projects to migrate to the newer ABI and continue to work "as
+well as they ever did". In the meantime, we can tighten up the rules
+around the "C" ABI.
+
+Simply taking that step, however, does not address the root problem of
+"de facto" dependencies. We've already seen that projects can and do
+rely on unwinding.
+
+The majority of this RFC is spent on describing the "C unwind" ABI and
+its details. We do note a few details about the "project group", but we
+expect the "project group" to largely define its own structure.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## The "C" ABI and unwinding
+
+Functions declared with the "C" ABI are not permitted to unwind.  In
+general, any attempt to unwind constitutes [Undefined Behavior], which
+means that the program may do arbitrary things.
```

Comment:
```suggestion
means that the program may do arbitrary things. Rust functions defined
with the "C" ABI are guaranteed to abort if an unwind attempts to
cross the function boundary.
```

Reply:
That's mentioned below, but I think it is ok to be more optimistic here and repeat that this RFC removes the amount of UB in Rust by defining some of it to abort.